src/v1/request-wrapper.ts

```diff
@@ -0,0 +1,44 @@
+
```

Comment:
This also we can bring out of V1

Reply:
The code is designed as Modular Design Pattern. So the dependencies should not kept outside or being shared.

For V2 we create a new folder and wrap everything. So we can break the dependency from v1 and up coming versions.

Reply:
I got it but why we have to repeat the code for requesting the API's
`async request(options:ReqOptions)` we can put outside and pass only versions for that so we don't want to repeat every versions right?